// @privy-io/api-types/resources/wallets/wallets.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as BalanceAPI from "./balance.mjs";
import { Balance } from "./balance.mjs";
import * as TransactionsAPI from "./transactions.mjs";
import { Transactions } from "./transactions.mjs";
import { Cursor } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Operations related to wallets
 */
export class Wallets extends APIResource {
    constructor() {
        super(...arguments);
        this.balance = new BalanceAPI.Balance(this._client);
        this.transactions = new TransactionsAPI.Transactions(this._client);
    }
    /**
     * Creates a new wallet on the requested chain and for the requested owner.
     *
     * @example
     * ```ts
     * const wallet = await client.wallets.create({
     *   chain_type: 'ethereum',
     * });
     * ```
     */
    create(params, options) {
        const { 'privy-idempotency-key': privyIdempotencyKey, ...body } = params;
        return this._client.post('/v1/wallets', {
            body,
            ...options,
            headers: buildHeaders([
                { ...(privyIdempotencyKey != null ? { 'privy-idempotency-key': privyIdempotencyKey } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Update a wallet's policies or authorization key configuration.
     *
     * @example
     * ```ts
     * const wallet = await client.wallets.update('wallet_id', {
     *   policy_ids: ['tb54eps4z44ed0jepousxi4n'],
     * });
     * ```
     */
    update(walletID, params, options) {
        const { 'privy-authorization-signature': privyAuthorizationSignature, 'privy-request-expiry': privyRequestExpiry, ...body } = params;
        return this._client.patch(path `/v1/wallets/${walletID}`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(privyAuthorizationSignature != null ?
                        { 'privy-authorization-signature': privyAuthorizationSignature }
                        : undefined),
                    ...(privyRequestExpiry != null ? { 'privy-request-expiry': privyRequestExpiry } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Get all wallets in your app.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const wallet of client.wallets.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v1/wallets', (Cursor), { query, ...options });
    }
    /**
     * Export a wallet's private key
     *
     * @example
     * ```ts
     * const walletExportResponseBody =
     *   await client.wallets.export('wallet_id', {
     *     encryption_type: 'HPKE',
     *     recipient_public_key:
     *       'BDAZLOIdTaPycEYkgG0MvCzbIKJLli/yWkAV5yCa9yOsZ4JsrLweA5MnP8YIiY4k/RRzC+APhhO+P+Hoz/rt7Go=',
     *   });
     * ```
     */
    export(walletID, params, options) {
        const { 'privy-authorization-signature': privyAuthorizationSignature, 'privy-request-expiry': privyRequestExpiry, ...body } = params;
        return this._client.post(path `/v1/wallets/${walletID}/export`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(privyAuthorizationSignature != null ?
                        { 'privy-authorization-signature': privyAuthorizationSignature }
                        : undefined),
                    ...(privyRequestExpiry != null ? { 'privy-request-expiry': privyRequestExpiry } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Get a wallet by wallet ID.
     *
     * @example
     * ```ts
     * const wallet = await client.wallets.get('wallet_id');
     * ```
     */
    get(walletID, options) {
        return this._client.get(path `/v1/wallets/${walletID}`, options);
    }
    /**
     * Sign a message with a wallet by wallet ID.
     *
     * @example
     * ```ts
     * const rawSignResponse = await client.wallets.rawSign(
     *   'wallet_id',
     *   {
     *     params: {
     *       bytes:
     *         '0a0234ea220809701d7a17a77e04408093e981a6335a66080112620a2d747970652e676f6f676c65617069732e636f6d2f70726f746f636f6c2e5472616e73666572436f6e747261637412310a15417009bf59e27d2031a23a61e1590289fc3d21b3cd121541132b98ed6fb80a2d45f177cdef091ae2d9dc115418e80770a0bee581a633',
     *       encoding: 'hex',
     *       hash_function: 'sha256',
     *     },
     *   },
     * );
     * ```
     */
    rawSign(walletID, params, options) {
        const { 'privy-authorization-signature': privyAuthorizationSignature, 'privy-idempotency-key': privyIdempotencyKey, 'privy-request-expiry': privyRequestExpiry, ...body } = params;
        return this._client.post(path `/v1/wallets/${walletID}/raw_sign`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(privyAuthorizationSignature != null ?
                        { 'privy-authorization-signature': privyAuthorizationSignature }
                        : undefined),
                    ...(privyIdempotencyKey != null ? { 'privy-idempotency-key': privyIdempotencyKey } : undefined),
                    ...(privyRequestExpiry != null ? { 'privy-request-expiry': privyRequestExpiry } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Sign a message or transaction with a wallet by wallet ID.
     *
     * @example
     * ```ts
     * const walletRpcResponse = await client.wallets.rpc(
     *   'wallet_id',
     *   {
     *     method: 'eth_sendTransaction',
     *     params: {
     *       transaction: {
     *         to: '0x0000000000000000000000000000000000000000',
     *         value: 1,
     *       },
     *     },
     *     chain_type: 'ethereum',
     *   },
     * );
     * ```
     */
    rpc(walletID, params, options) {
        const { 'privy-authorization-signature': privyAuthorizationSignature, 'privy-idempotency-key': privyIdempotencyKey, 'privy-request-expiry': privyRequestExpiry, ...body } = params;
        return this._client.post(path `/v1/wallets/${walletID}/rpc`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(privyAuthorizationSignature != null ?
                        { 'privy-authorization-signature': privyAuthorizationSignature }
                        : undefined),
                    ...(privyIdempotencyKey != null ? { 'privy-idempotency-key': privyIdempotencyKey } : undefined),
                    ...(privyRequestExpiry != null ? { 'privy-request-expiry': privyRequestExpiry } : undefined),
                },
                options?.headers,
            ]),
        });
    }
}
Wallets.Balance = Balance;
Wallets.Transactions = Transactions;
