// @privy-io/react-auth/dist/cjs/useActiveWallet-gmIKm4kj.js
"use strict";var e=require("react"),t=require("styled-components"),r=require("react/jsx-runtime"),o=require("./context-DGXRy2ro.js"),n=require("./events-context-Di6--rDg.js"),i=require("./wallet-connect-DxvWg0_1.js"),l=require("./internal-context-BJv4f_bO.js"),s=require("./privy-context-v_nKCz1o.js"),a=require("./index-78u8hYh9.js"),d=require("./get-entropy-details-for-user-pjIIkVsN.js"),c=require("zustand");const p=t.styled.div`
  text-align: left;
  flex-grow: 1;
`,u=t.styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  flex-grow: 1;
`,x=t.styled.div`
  display: flex;
  flex-direction: column;
  gap: 8px;

  /* for Internet Explorer, Edge */
  -ms-overflow-style: none;

  /* for Firefox */
  scrollbar-width: none;

  /* for Chrome, Safari, and Opera */
  &::-webkit-scrollbar {
    display: none;
  }
`,y=t.styled(x)`
  ${e=>"light"===e.$colorScheme?"background: linear-gradient(var(--privy-color-background), var(--privy-color-background) 70%) bottom, linear-gradient(rgba(0, 0, 0, 0) 20%, rgba(0, 0, 0, 0.06)) bottom;":"dark"===e.$colorScheme?"background: linear-gradient(var(--privy-color-background), var(--privy-color-background) 70%) bottom, linear-gradient(rgba(255, 255, 255, 0) 20%, rgba(255, 255, 255, 0.06)) bottom;":void 0}

  background-repeat: no-repeat;
  background-size:
    100% 32px,
    100% 16px;
  background-attachment: local, scroll;
  max-height: 400px;
  overflow-y: auto;
  scrollbar-width: none;
  padding: 3px;
`;let g=t.css`
  && {
    width: 100%;
    font-size: 16px;
    line-height: 24px;
    min-height: 56px;

    /* Tablet and Up */
    @media (min-width: 440px) {
      font-size: 14px;
    }

    display: flex;
    gap: 12px;
    align-items: center;
    color: var(--privy-color-foreground);

    padding: 10px 12px;
    border: 1px solid var(--privy-color-foreground-4) !important;
    border-radius: var(--privy-border-radius-md);
    transition: background-color 200ms ease;

    cursor: pointer;

    &:hover {
      background-color: var(--privy-color-background-2);
    }

    &:disabled {
      cursor: pointer;
      background-color: var(--privy-color-background-2);
    }
  }
`;const v=t.styled.div`
  text-align: center;
  font-size: 14px;
  margin-bottom: 24px;
`,f=t.styled.button.attrs({className:"login-method-button"})`
  ${g}
`;t.styled.a`
  ${g}
`;const h=t.styled.div`
  width: 32px;
  height: 32px;
  border-radius: ${e=>e.$fullSize?"0":"4px"};
  background: ${e=>e.$fullSize?"transparent":"var(--privy-color-background-2)"};
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;

  svg {
    width: ${e=>e.$fullSize?"32px":"18px"};
    height: ${e=>e.$fullSize?"32px":"18px"};
    color: ${e=>e.$fullSize?"inherit":"var(--privy-color-icon-muted)"};
  }
`,b=t.styled.div`
  width: 100%;
  height: 100%;
  min-height: inherit;
  display: flex;
  flex-direction: column;
  ${e=>e.$if?"display: none;":""}
`,m=t.styled.div`
  width: 100%;
  height: 100%;
  padding: ${e=>e.$withPadding?"64px 0px":"0px"};
`,w=t.styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-bottom: 32px;
  gap: 12px;
  & h3 {
    font-size: 18px;
    font-style: normal;
    font-weight: 600;
    line-height: 24px;
  }
  & p {
    max-width: 300px;
    font-size: 14px;
    font-style: normal;
    font-weight: 400;
    line-height: 20px;
  }
`;let S=t.styled.span`
  && {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 2;
  }
`,$=t.styled.span`
  position: relative;
  width: 82px;
  height: 82px;
  display: inline-block;
`;const k=t.styled.span`
  && {
    width: 82px;
    height: 82px;
    border-width: 4px;
    border-style: solid;
    border-color: ${e=>e.color??"var(--privy-color-icon-subtle)"};
    border-bottom-color: transparent;
    border-radius: 50%;
    display: inline-block;
    box-sizing: border-box;
    animation: rotation 1.2s linear infinite;
    transition: border-color 800ms;
  }

  @keyframes rotation {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }

  &&&.success {
    border-color: var(--privy-color-icon-success);
    border-bottom-color: var(--privy-color-icon-success);
  }

  &&&.fail {
    border-color: var(--privy-color-icon-error);
    border-bottom-color: var(--privy-color-icon-error);
  }
`,C=t.styled(k)`
  && {
    border-bottom-color: ${e=>e.color??"var(--privy-color-border-default)"};
    border-color: ${e=>e.color??"var(--privy-color-border-default)"};
    animation: none;
    opacity: 0.5;
  }
`;let j=t.styled(k)`
  && {
    height: 1rem;
    width: 1rem;
    margin: 2px 0;
    border-width: 1.5px;

    /* Override default Loader to match button transitions */
    transition: border-color 200ms ease;
  }
`,W=/*#__PURE__*/e.createContext({});let z=e=>`privy:${e}:recent-login-method`,T=e=>`privy:${e}:recent-login-wallet-client`,P=e=>`privy:${e}:recent-login-chain-type`;const M=c.create((()=>({isModalOpen:!1,resolvers:null})));let E=c.create((()=>({})));const L=async()=>{let e,t,r=new Promise(((r,o)=>{e=r,t=o}));return E.setState({}),M.setState({isModalOpen:!0,resolvers:{resolve:e,reject:t}}),await r},q=({address:e,client:t,appId:r})=>{let o=`${t}:${e}`;r&&i.s.put(A(r),o),M.setState({wallet:o})},A=e=>`privy:${e}:active-wallet-connection`;exports.AlignBottom=u,exports.ButtonLoader=e=>/*#__PURE__*/r.jsx(j,{color:e.color||"var(--privy-color-foreground-3)"}),exports.ConnectionLoader=({success:e,fail:t})=>/*#__PURE__*/r.jsxs($,{children:[/*#__PURE__*/r.jsx(S,{children:/*#__PURE__*/r.jsx(C,{className:e?"success":t?"fail":""})}),/*#__PURE__*/r.jsx(S,{children:/*#__PURE__*/r.jsx(k,{className:e?"success":t?"fail":""})})]}),exports.EmptyWalletState=m,exports.Grow=p,exports.Header=w,exports.Hide=b,exports.Loader=k,exports.LoaderFaint=C,exports.LoginMethodButton=f,exports.LoginMethodContainer=x,exports.LoginMethodContainerWithScrollShadows=y,exports.LoginMethodIconWrapper=h,exports.RecentlyUsedAccountProvider=({children:t})=>{let l=o.useAppConfig(),[s,a]=e.useState({});return n.usePrivyEventSubscription("login",{onComplete:({loginAccount:e})=>{e&&"passkey"!==e.type&&"cross_app"!==e.type&&("wallet"!==e.type||"privy"!==e.walletClientType)&&(i.s.put(z(l.id),e.type),"wallet"===e.type?(i.s.put(T(l.id),e.walletClientType),i.s.put(P(l.id),e.chainType),a({accountType:e.type,walletClientType:e.walletClientType,chainType:e.chainType})):(i.s.del(T(l.id)),i.s.del(P(l.id)),a({accountType:e.type})))}}),e.useEffect((()=>{if(!l.id)return;let e=i.s.get(z(l.id)),t=i.s.get(T(l.id)),r=i.s.get(P(l.id));e&&a("wallet"===e?{accountType:e,walletClientType:t,chainType:r}:{accountType:e})}),[l.id]),/*#__PURE__*/r.jsx(W.Provider,{value:s,children:t})},exports.Subtitle=v,exports.close=()=>{let e=M.getState().resolvers;e||console.warn("useActiveWalletStore: Called close while modal was not open"),M.setState({isModalOpen:!1}),e?.resolve({wallet:M.getState().wallet})},exports.setActiveWallet=q,exports.updateBalance=(e,t,r)=>{E.setState((o=>({[e]:{...o[e],[t]:r}})))},exports.useActiveWallet=()=>{let t=o.useAppConfig(),{wallets:r}=d.useWallets(),{connectors:n}=l.usePrivyInternal(),s=e.useMemo((()=>n.filter(a.isSolanaWalletConnector).flatMap((e=>e.wallets))),[n]),c=M((e=>e.wallet));e.useEffect((()=>{if(!t.id)return;let e=i.s.get(A(t.id));M.setState({wallet:e})}),[t.id]);let p=e.useMemo((()=>{if(!c)return;let[,e]=c.split(":");return r.find((t=>t.address===e))||s.find((t=>t.address===e))}),[c,r,s]),u=e.useMemo((()=>p?"ethereum"===p.type?p.chainId:"solana":void 0),[p]);return{connect:async e=>{e?.reset&&(i.s.del(A(t.id)),M.setState({wallet:void 0}));let{wallet:o}=await L();if(!o)return{};let[,n]=o.split(":"),l=r.find((e=>e.address===n));if(l)return{wallet:l,network:l.chainId};let a=s.find((e=>e.address===n));return a?{wallet:a,network:"solana"}:{}},setActiveWallet:e.useCallback((e=>{q({address:e.address,client:e.walletClientType,appId:t.id})}),[t.id]),wallet:p,network:u}},exports.useBalance=(e,t)=>E((r=>r[e]?.[t])),exports.useConnectWallet=function(t){let{connectWallet:r}=e.useContext(s.PrivyContext);return n.usePrivyEventSubscription("connectWallet",t),{connectWallet:r}},exports.useElementHeight=t=>{let[r,o]=e.useState("auto");return e.useEffect((()=>{let e=new ResizeObserver((e=>{o(e[0]?.contentRect.height??"auto")}));return t.current&&e.observe(t.current),()=>{t.current&&e.unobserve(t.current)}}),[t.current]),r},exports.useFundWallet=e=>{n.usePrivyEventSubscription("fundWallet",e);let{fundWallet:t}=l.usePrivyInternal();return{fundWallet:({address:e,options:r})=>t(e,r)}},exports.useIsModalOpen=()=>M((e=>e.isModalOpen)),exports.useLogin=function(t){let{login:r}=e.useContext(s.PrivyContext);return n.usePrivyEventSubscription("login",t),{login:r}},exports.useLogout=function(t){let{logout:r}=e.useContext(s.PrivyContext);return n.usePrivyEventSubscription("logout",t),{logout:r}},exports.useRecentlyUsedLogin=()=>e.useContext(W);
